trenchboot/02-with-drtm.robot

```diff
@@ -86,19 +86,70 @@ WTD005.001 DRTM event log has DRTM entries (AMD)
     ...    find_event 17 "DLME"
     Should Be Equal As Strings    ${match}    MATCH    No "DLME" entry
 
+    # XXX: No part of SLRT is measured on AMD platforms.
     # Early TPM code doesn't identify SLRT measurement event in any way.
-    ${match}=    Execute Command In Terminal
-    ...    find_event 18 ""
-    Should Be Equal As Strings    ${match}    MATCH    No SLRT entry
+    # ${match}=    Execute Command In Terminal
+    # ...    find_event 18 ""
+    # Should Be Equal As Strings    ${match}    MATCH    No SLRT entry
 
-    ${match}=    Execute Command In Terminal
-    ...    find_event 17 "Measured MB2 module"
-    Should Be Equal As Strings    ${match}    MATCH\nMATCH    No "Measured MB2 module" entries
+    # XXX: This is only for Multiboot2 protocol.
+    # ${match}=    Execute Command In Terminal
+    # ...    find_event 17 "Measured MB2 module"
+    # Should Be Equal As Strings    ${match}    MATCH\nMATCH    No "Measured MB2 module" entries
 
 WTD006.001 DRTM log aligns with PCR values
     [Documentation]    Verify that all DRTM measurements are correctly reflected
     ...    in DRTM event log
-    Validate PCRs Against Event Log    /sys/kernel/security/slaunch/eventlog
+    Execute Command In Terminal
+    ...    tpm2_eventlog /sys/kernel/security/slaunch/eventlog > /tmp/event-log
+    # This extracts hashes for a combination of a PCR (first parameter) and an
+    # algorithm (second parameter).  The printed line have the form
+    #    {event data length}-{digest}
+    # to make it possible to ignore some events based on a combination of these
+    # fields.
+    #
+    # Backslash in front of some dollar signs is escaping AWK fields while
```

Comment:
This seems very convoluted.

Perhaps we shall we include such helper shell scripts in this repo, send them to DUT and just execute in the test suite?

Reply:
Then we would have to have SSH and know the device IP. RF really isn't made for such tests, that's why I didn't fully understand the decision to remove such tests from cukinia.

Reply:
> Then we would have to have SSH and know the device IP.

I think we do have it. And we do not need to know DUT IP, this should be automatically retrieved by the keywords if we have shell over serial. I think we can also send a file over serial, if no SSH is available as well.

> RF really isn't made for such tests, that's why I didn't fully understand the decision to remove such tests from cukinia.

If you're referring to: https://github.com/3mdeb/meta-trenchboot/pull/33#discussion_r1666680491 then I proposed some `rule of thumb` that made sense to me at the time. If there were strong opinions on other proposal, I may have missed them. We need to make decisions to move forward. If I made a bad one, I will be happily proven wrong and change it if it will make our lives better.

Ultimately, we want to execute some commands in RF as test executor, as we have control over hardware there. Unless you have other suggestions here as well.

Reply:
Would it make sense to have such scripts as part of meta-trenchboot? It could potentially make it harder to run the tests in final environment (i.e. kernel of some widely used distro), but we don't plan on running tests there in the foreseeable future. Unfortunately, it could also mean that modifications to tests would require changes in two separate repositories.

Reply:
> I can feel the pain while wrapping bash functions inside this RF code.

Every time I deal with RF, it reminds me of that [worst programming language ever talk](https://www.youtube.com/watch?v=vcFBwt1nu2U) (although I watched an older version).  And because *everything* about RF is pain, this doesn't really standout.  Avoiding any reboots of the DUT on subsequent runs also helps making such things work.

Adding local variables for parameters might help in case of this particular function, because there will be no mixing of `$1` from both bash and awk.  Another possibility is to move this to Python and use its more adequate raw string literals, possibly with some format strings.  Although, just like with separate script files, this will mean that different parts of the test will be in different places.

Reply:
If you guys have suggestions on how to make our lives less miserable, I am all ears. We can create a ticket for discussion if needed.

Reply:
We have concluded with @krystian-hebel that we leave it as is since it works. 

Place to discuss it in the future: https://github.com/Dasharo/open-source-firmware-validation/issues/573